### PR TITLE
http1 support in e2e test request signing

### PR DESCRIPTION
**Pull request by mxfactorial**
*Issue #, if available:*

*Description of changes:*

https://github.com/awslabs/aws-sdk-rust/issues/1041 enables switching [apply_to_request_http0x](https://github.com/awslabs/aws-lambda-web-adapter/blob/e51f6557420f5817aadf6f98356df882d7ccd879/tests/e2e_tests/main.rs#L95) to [apply_to_request_http1x](https://docs.rs/aws-sigv4/latest/aws_sigv4/http_request/struct.SigningInstructions.html#method.apply_to_request_http1x)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by bnusunny** [comment]
Thanks!

File: Cargo.toml [hunk 1]
```diff
@@ -47,7 +52,7 @@ httpmock = "0.7.0"
 http-body-util = "0.1.0"
 http-body = "1.0"
 hyper-rustls = "0.26"
-aws-sigv4 = { version = "1.1" }
+aws-sigv4 = { version = "1.1.6", features = ["http1"] }
```
Comment (bnusunny) [inline Cargo.toml:55]: 'http1' is enabled with default feature. No need to list it again.

File: tests/e2e_tests/util.rs
Comment (bnusunny) [inline tests/e2e_tests/util.rs:1]: This whole file should be removed. It is for the workaround.

**Review by bnusunny — CHANGES_REQUESTED** [review]

**Comment by bnusunny** [comment]
I have a few comments. Could you take a look?

**Comment by bnusunny** [comment]
And please also fix the lint issue

**Comment by mxfactorial** [comment]
ready

```
❯ make fmt && make lint && make test
cargo fmt --all -- --check
cargo clippy -- -Dwarnings
    Checking lambda_web_adapter v0.8.1 (/Users/mx/code/aws-lambda-web-adapter)
    Finished dev [unoptimized + debuginfo] target(s) in 0.65s
cargo nextest run
warning: ignoring unknown configuration keys in config file /Users/mx/code/aws-lambda-web-adapter/.config/nextest.toml: profile.ci.inherits
   Compiling lambda_web_adapter v0.8.1 (/Users/mx/code/aws-lambda-web-adapter)
    Finished test [unoptimized + debuginfo] target(s) in 3.13s
    Starting 18 tests across 4 binaries (4 skipped)
        PASS [   0.251s] lambda_web_adapter tests::test_status_200_is_ok
        PASS [   0.249s] lambda_web_adapter tests::test_status_500_is_bad
        PASS [   0.244s] lambda_web_adapter::integ_tests test_http_content_encoding_suffix
        PASS [   0.247s] lambda_web_adapter::integ_tests test_http_basic_request
        PASS [   0.246s] lambda_web_adapter::integ_tests test_http_compress
        PASS [   0.249s] lambda_web_adapter::integ_tests test_adapter_options_from_env
        PASS [   0.245s] lambda_web_adapter::integ_tests test_http_compress_disallowed_type
        PASS [   0.246s] lambda_web_adapter::integ_tests test_http_compress_already_compressed
        PASS [   0.248s] lambda_web_adapter::integ_tests test_adapter_options_from_namespaced_env
        PASS [   0.250s] lambda_web_adapter tests::test_status_403_is_bad_when_configured
        PASS [   0.223s] lambda_web_adapter::integ_tests test_http_context_headers
        PASS [   0.086s] lambda_web_adapter::integ_tests test_http_context_multi_headers
        PASS [   0.070s] lambda_web_adapter::integ_tests test_http_headers
        PASS [   0.086s] lambda_web_adapter::integ_tests test_http_path_encoding
        PASS [   0.069s] lambda_web_adapter::integ_tests test_http_post_put_delete
        PASS [   0.059s] lambda_web_adapter::integ_tests test_http_query_params
        PASS [   0.059s] lambda_web_adapter::integ_tests test_http_readiness_check
        PASS [   0.055s] lambda_web_adapter::integ_tests test_readiness_check_port_fallback_to_lwa_port
------------
     Summary [   0.323s] 18 tests run: 18 passed, 4 skipped
```

**Review by bnusunny — APPROVED** [review]

**Pull request merged**